src/main/scala/ai/privado/rulevalidator/YamlFileValidator.scala

```diff
@@ -164,20 +173,40 @@ object YamlFileValidator {
     ruleFile: File,
     ruleJsonTree: JsonNode,
     callerCommand: String = ""
-  ): Either[Unit, String] = {
+  ): Either[Unit, (String, String)] = {
     val configTypeKey =
       if (ruleJsonTree.fieldNames().hasNext) ruleJsonTree.fieldNames().next() else CatLevelOne.UNKNOWN.name
     ConfigRuleType.withNameDefaultHandler(configTypeKey) match {
-      case ConfigRuleType.SEMANTICS      => Right(SEMANTICS)
-      case ConfigRuleType.EXCLUSIONS     => Right(EXCLUSIONS)
-      case ConfigRuleType.SINK_SKIP_LIST => Right(SINK_SKIP_LIST)
+      case ConfigRuleType.SEMANTICS      => Right(ConfigRuleType.SEMANTICS.toString, SEMANTICS)
+      case ConfigRuleType.EXCLUSIONS     => Right(ConfigRuleType.EXCLUSIONS.toString, EXCLUSIONS)
+      case ConfigRuleType.SINK_SKIP_LIST => Right(ConfigRuleType.SINK_SKIP_LIST.toString, SINK_SKIP_LIST)
       case _ =>
         if (callerCommand == CommandConstants.VALIDATE) println(PRETTY_LINE_SEPARATOR)
         println(f"File : ${ruleFile.pathAsString}: Config Invalid. Correct the rules and try again.")
         Left(())
     }
   }
 
+  def validateRegexPattern(content: JsonNode, typeOfRule: String): Either[mutable.Set[String], Unit] = {
+    val contentJson = ujson.read(content.toString)(typeOfRule).arr.toList
+    val errors = mutable.Set[String]()
+    contentJson.foreach(value => {
+      logger.debug(value("patterns").arr.toList.toString())
```

Comment:
@abhstabs This logger statement will flood the log with regex's, let's log only the incorrect ones